src/cclib/parser/gamessparser.py

```diff
@@ -203,38 +203,44 @@ def extract(self, inputfile, line):
         #                       E(2)=        -0.1403745370
         #                     E(MP2)=       -76.1492222841
         #
+        # With GAMESS-US 20 APR 2017 (R1), the following block may be present:
+        #       SCHWARZ INEQUALITY TEST SKIPPED          0 INTEGRAL BLOCKS
+        # ... END OF INTEGRAL TRANSFORMATION ...
+
         if line.find("RESULTS OF MOLLER-PLESSET") >= 0 or line[6:37] == "SCHWARZ INEQUALITY TEST SKIPPED":
 
             if not hasattr(self, "mpenergies"):
                 self.mpenergies = []
 
+            line = next(inputfile)
             # Each iteration has a new print-out
-            self.mpenergies.append([])
-
-            # GAMESS-US presently supports only second order corrections (MP2)
-            # PC GAMESS also has higher levels (3rd and 4th), with different output
-            # Only the highest level MP4 energy is gathered (SDQ or SDTQ)
-            while re.search("DONE WITH MP(\d) ENERGY", line) is None:
-
-                line = next(inputfile)
-                if len(line.split()) > 0:
-                    # Only up to MP2 correction
-                    if line.split()[0] == "E(MP2)=":
-                        self.metadata["methods"].append("MP2")
-                        mp2energy = float(line.split()[1])
-                        self.mpenergies[-1].append(utils.convertor(mp2energy, "hartree", "eV"))
-                    # MP2 before higher order calculations
-                    if line.split()[0] == "E(MP2)":
-                        mp2energy = float(line.split()[2])
-                        self.mpenergies[-1].append(utils.convertor(mp2energy, "hartree", "eV"))
-                    if line.split()[0] == "E(MP3)":
-                        self.metadata["methods"].append("MP3")
-                        mp3energy = float(line.split()[2])
-                        self.mpenergies[-1].append(utils.convertor(mp3energy, "hartree", "eV"))
-                    if line.split()[0] in ["E(MP4-SDQ)", "E(MP4-SDTQ)"]:
-                        self.metadata["methods"].append("MP4")
-                        mp4energy = float(line.split()[2])
-                        self.mpenergies[-1].append(utils.convertor(mp4energy, "hartree", "eV"))
+            if re.search("END OF INTEGRAL TRANSFORMATION", line) is None:
```

Comment:
Any reason for using ```re.search()``` instead of ```blah in foo``` or ```foo.find(blah) >= 0```? Not that it really matters, but it's *much* slower:

```
[adam_tenderholt@iMac:~/Downloads]$ python3 -m timeit -s "import re" "re.search('END OF INTEGRAL TRANSFORMATION', 'blah foo bar')"
1000000 loops, best of 3: 0.641 usec per loop
[adam_tenderholt@iMac:~/Downloads]$ python3 -m timeit -s "'END OF INTEGRAL TRANSFORMATION' in 'blah foo bar'"
100000000 loops, best of 3: 0.00842 usec per loop
[adam_tenderholt@iMac:~/Downloads]$ python3 -m timeit -s "'blah foo bar'.find('END OF INTEGRAL TRANSFORMATION') >= 0"
100000000 loops, best of 3: 0.00801 usec per loop
```